build/common.props

```diff
@@ -16,4 +16,12 @@
     <VersionSuffix Condition="'$(VersionSuffix)'!='' AND '$(BuildNumber)' != ''">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
   </PropertyGroup>
 
-</Project>
\ No newline at end of file
+  <!-- resolve using git.exe because the GetGitCommitInfoTask can't be loaded before it is compiled -->
+  <!-- TODO set this in repo.targets so it applies to the entire build -->
+  <Target Name="ResolveCommitHashWithExe" Condition="'$(CommitHash)'==''" BeforeTargets="PrepareForBuild">
+    <Exec Command="git rev-parse HEAD" ConsoleToMSBuild="true" EchoOff="true" Condition="'$(CommitHash)'==''">
```

Comment:
If we need to do this in places where we can't use a custom task, it might just be worth using git.exe everywhere. The main reason I didn't do that was because I didn't want our build to be dependent upon git being in the path.